File: mobile/lib/auth/BuyerSignUpRequest.dart
import 'dart:convert';

class BuyerSignupRequest {
  final String username;
  final String email;
  final String password;
  final String name;
  final List<String> preferredDeliveryMethods;

  BuyerSignupRequest({
    required this.username,
    required this.email,
    required this.password,
    required this.name,
    required this.preferredDeliveryMethods,
  });

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'email': email,
      'password': password,
      'name': name,
      'preferredDeliveryMethods': preferredDeliveryMethods,
    };
  }
}

File: src/main/java/com/jgroup/farmers_market/model/api/BuyerSignupRequest.java
package com.jgroup.farmers_market.model.api;

import com.jgroup.farmers_market.model.enums.EDeliveryMethod;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class BuyerSignupRequest {
    private String username;
    private String email;
    private String password;
    private String name;
    private Set<EDeliveryMethod> preferredDeliveryMethods;
}
